spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/GatewayMetricsFilter.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.gateway.filter;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.core.Ordered;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ServerWebExchange;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import reactor.core.publisher.Mono;
+
+public class GatewayMetricsFilter implements GlobalFilter, Ordered {
+
+	private MeterRegistry meterRegistry;
+
+	public GatewayMetricsFilter(MeterRegistry meterRegistry) {
+		this.meterRegistry = meterRegistry;
+	}
+
+	@Override
+	public int getOrder() {
+		return NettyWriteResponseFilter.WRITE_RESPONSE_FILTER_ORDER - 1;
+	}
+
+	@Override
+	public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+		long start = System.nanoTime();
+		return chain.filter(exchange).then(Mono.defer(() -> {
+			HttpStatus statusCode = exchange.getResponse().getStatusCode();
+			boolean success = statusCode.is2xxSuccessful();
```

Comment:
Are 3xx statuses considered successful as well? There is a proposal for status code group https://github.com/spring-projects/spring-boot/issues/13801 that differentiates between client and server errors as well. Might want to be consistent with that.

Reply:
Maybe change the tag from success to is2xxSuccessful since the status tag will also be available?

Reply:
I think it maybe beneficial to follow https://github.com/spring-projects/spring-boot/issues/13801.

Reply:
You mean drop the success tag and introduce 2xx, 4xx and 5xx as tags?

Reply:
Done. Removed success tag and added outcome as described.